### ISSUE TITLE
"Compilation error in v3.0.1."

### ISSUE THREAD

**Issue by hamidMech** [Exawind/amr-wind#1231]
## Bug description
A compilation error occurs in v-3.0.1.

## Steps to reproduce

1. Compiler used
   - [x] GCC
   - [ ] LLVM
   - [ ] oneapi (Intel)
   - [ ] nvcc (NVIDIA)
   - [ ] rocm (AMD)
   - [x] with MPI
   - [ ] other:
2. Operating system
   - [x] Linux
   - [ ] OSX
   - [ ] Windows
   - [ ] other (do tell ;)):
3. Hardware:
   - [x] CPU
   - [ ] GPU
4. Machine details ():
```
[user@hpc_cluster build]$ module list

Currently Loaded Modules:
  1) hpc/.1.9.5                             (H,S)   4) binutils/.2.38 (H)   7) libevent/2.1.12      10) zlib/.1.2.12      (H)  13) hwloc/.2.7.1 (H)  16) UCC/1.0.0        19) FlexiBLAS/3.2.0  22) ScaLAPACK/2.2.0-fb     25) CMake/3.27.9
  2) buildtool-easybuild/4.8.0-hpce082752a2         5) GCC/11.3.0           8) numactl/.2.0.14 (H)  11) libxml2/2.9.13         14) PMIx/4.1.2        17) OpenMPI/4.1.4    20) FFTW/3.3.10      23) foss/2022a             26) netCDF-HDF5/4.9.2-1.12.2-hpc1
  3) GCCcore/11.3.0  
```
5. Input file attachments 
```
export OpenFAST_DIR=/home/user/myProgram/openfast-v3.5.3/install

git clone --recursive https://github.com/exawind/amr-wind.git

mkdir amr-wind-v3.0.1/build
cd amr-wind-v3.0.1/build

cmake -DAMR_WIND_ENABLE_TESTS:BOOL=ON -DAMR_WIND_ENABLE_MPI:BOOL=ON -DAMR_WIND_ENABLE_NETCDF:BOOL=ON -DAMR_WIND_ENABLE_OPENFAST:BOOL=ON -DAMR_WIND_ENABLE_OPENMP:BOOL=ON -DAMR_WIND_ENABLE_HDF5:BOOL=ON -DCMAKE_INSTALL_PREFIX=/home/user/myProgram/amr-wind-v3.0.1/install ../

make -j10
```
6. Error (paste or attach):
```
.
.
.
[ 35%] Built target amrex_hydro
[ 35%] Linking CXX static library libamrex_hydro_api.a
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_advance.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_compute_dt.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_regrid.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/CFDSim.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/helics.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/SimTime.cpp.o
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/IntField.cpp.o
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/Field.cpp.o
[ 38%] Built target amrex_hydro_api
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/FieldRepo.cpp.o
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/ScratchField.cpp.o
In file included from /home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/Field.cpp:5:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In member function ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>)’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:438:40: error: ‘mfab’ was not declared in this scope; did you mean ‘mfabs’?
  438 |                 for (amrex::MFIter mfi(mfab); mfi.isValid(); ++mfi) {
      |                                        ^~~~
      |                                        mfabs
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/IntScratchField.cpp.o
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In instantiation of ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>) [with BCOpCreator = amr_wind::FieldBCNoOp; amrex::Real = double; amrex::Array<amrex::MultiFab*, 3> = std::array<amrex::MultiFab*, 3>]’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:406:10:   required from here
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:437:23: warning: unused variable ‘mfab’ [-Wunused-variable]
  437 |                 auto& mfab = *mfabs[fdir];
      |                       ^~~~
make[2]: *** [CMakeFiles/amrwind_obj.dir/build.make:174: CMakeFiles/amrwind_obj.dir/amr-wind/core/Field.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1607: CMakeFiles/amrwind_obj.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```
8. If this is a segfault, a stack trace from a debug build (paste or attach):
```
N/A
```

## Expected behavior
```
The compilation is continuing.
```
## AMR-Wind information

```
N/A
```

## Additional context
```
N/A
```

**Comment by marchdf** [comment]
hmmm I can't reproduce this and the CI and nightlies should/would have caught this. 

Can you do me a favor and try a simpler build on your system? 
```
cmake -DAMR_WIND_ENABLE_TESTS:BOOL=OFF -DAMR_WIND_ENABLE_MPI:BOOL=OFF -DAMR_WIND_ENABLE_NETCDF:BOOL=OFF -DAMR_WIND_ENABLE_OPENFAST:BOOL=OFF -DAMR_WIND_ENABLE_OPENMP:BOOL=OFF -DAMR_WIND_ENABLE_HDF5:BOOL=OFF -DCMAKE_INSTALL_PREFIX=/home/user/myProgram/amr-wind-v3.0.1/install ../
```

If that works, then gradually turn those flags back on and see what happens?

**Comment by hamidMech** [comment]
Thanks for the hint. The error is because of "-DAMR_WIND_ENABLE_OPENMP:BOOL=ON". None of the other flags trigger a compilation error when enabled (ON).

In addition, I have compiled openfast-v3.5.3 with 

cmake .. -DFPE_TRAP_ENABLED=ON -DBUILD_OPENFAST_CPP_API:BOOL=ON -DOPENMP=ON -DDOUBLE_PRECISION=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_FASTFARM:BOOL=ON -DOPENMP=ON 

Here coms the error of compilation:
-----------------------------------
cmake -DAMR_WIND_ENABLE_TESTS:BOOL=OFF -DAMR_WIND_ENABLE_MPI:BOOL=OFF -DAMR_WIND_ENABLE_NETCDF:BOOL=OFF -DAMR_WIND_ENABLE_OPENFAST:BOOL=OFF -DAMR_WIND_ENABLE_OPENMP:BOOL=ON -DAMR_WIND_ENABLE_HDF5:BOOL=OFF -DCMAKE_INSTALL_PREFIX=/home/user/myProgram/amr-wind-v3.0.1/install ../
.
.
.
[ 34%] Built target amrex_hydro
[ 35%] Linking CXX static library libamrex_hydro_api.a
[ 35%] Built target amrex_hydro_api
[ 35%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_advance.cpp.o
[ 35%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_regrid.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/incflo_compute_dt.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/helics.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/CFDSim.cpp.o
[ 36%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/ViewField.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/SimTime.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/Field.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/IntField.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/FieldRepo.cpp.o
[ 37%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/IntScratchField.cpp.o
[ 38%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/ScratchField.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/MeshMap.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/core/MLMGOptions.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/FixedGradientBC.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/BCInterface.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/convection/incflo_mol_fluxes.cpp.o
[ 39%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/scalar_bcs.cpp.o
[ 40%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/MassInflowOutflowBC.cpp.o
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/wall_models/WallFunction.cpp.o
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/projection/incflo_apply_nodal_projection.cpp.o
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/diffusion/incflo_diffusion.cpp.o
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/setup/init.cpp.o
In file included from /home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/Field.cpp:5:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In member function ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>)’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:438:40: error: ‘mfab’ was not declared in this scope; did you mean ‘mfabs’?
  438 |                 for (amrex::MFIter mfi(mfab); mfi.isValid(); ++mfi) {
      |                                        ^~~~
      |                                        mfabs
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/diagnostics.cpp.o
In file included from /home/user/myProgram/amr-wind-v3.0.1/amr-wind/boundary_conditions/scalar_bcs.H:5,
                 from /home/user/myProgram/amr-wind-v3.0.1/amr-wind/boundary_conditions/scalar_bcs.cpp:1:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In member function ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>)’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:438:40: error: ‘mfab’ was not declared in this scope; did you mean ‘mfabs’?
  438 |                 for (amrex::MFIter mfi(mfab); mfi.isValid(); ++mfi) {
      |                                        ^~~~
      |                                        mfabs
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In instantiation of ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>) [with BCOpCreator = amr_wind::BCOpCreator<amr_wind::udf::CustomScalar, amr_wind::ConstDirichlet>; amrex::Real = double; amrex::Array<amrex::MultiFab*, 3> = std::array<amrex::MultiFab*, 3>]’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:406:10:   required from here
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:437:23: warning: unused variable ‘mfab’ [-Wunused-variable]
  437 |                 auto& mfab = *mfabs[fdir];
      |                       ^~~~
[ 41%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/io.cpp.o
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H: In instantiation of ‘void amr_wind::FieldFillPatchOps<BCOpCreator>::set_inflow_sibling_fields(int, amrex::Real, amrex::Array<amrex::MultiFab*, 3>) [with BCOpCreator = amr_wind::FieldBCNoOp; amrex::Real = double; amrex::Array<amrex::MultiFab*, 3> = std::array<amrex::MultiFab*, 3>]’:
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:406:10:   required from here
/home/user/myProgram/amr-wind-v3.0.1/amr-wind/core/FieldFillPatchOps.H:437:23: warning: unused variable ‘mfab’ [-Wunused-variable]
  437 |                 auto& mfab = *mfabs[fdir];
      |                       ^~~~
[ 42%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/bc_ops.cpp.o
[ 42%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/console_io.cpp.o
[ 42%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/index_operations.cpp.o
[ 42%] Building CXX object CMakeFiles/amrwind_obj.dir/amr-wind/utilities/IOManager.cpp.o
make[2]: *** [CMakeFiles/amrwind_obj.dir/build.make:314: CMakeFiles/amrwind_obj.dir/amr-wind/boundary_conditions/scalar_bcs.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/amrwind_obj.dir/build.make:174: CMakeFiles/amrwind_obj.dir/amr-wind/core/Field.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1525: CMakeFiles/amrwind_obj.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by marchdf** [comment]
OK I can reproduce and hopefully can fix soon. Thank you for bringing this up. I would caution against openmp in openfast with amr-wind. I am not sure how well tested that pathway is. Just make sure to triple check your results ;)
